### DISCUSSION TITLE
Authentication with MongoDB

### DISCUSSION THREAD

**Discussion by cassiofrancklin** [adonisjs/core#2326] [Help]
Hey, I'm starting a project that should use Adonis.js and MongoDB, I know Adonis doesn't support MongoDB by default through Lucid library, so I'm going to use Mongoose to handle things related to database.

After that, is there any way to integrate Mongo and the standard authentication methods that Adonis provide?
I think it wouldn't be wise to import passport, jwt etc. and create my own auth system, since Adonis has these authentication tools by default.

I would like to do something like this:

`  jwt: {
    serializer: "mongoose",
    model: "App/Models/user.schema",
    scheme: "jwt",
    uid: "email",
    password: "password",
    options: {
      secret: Env.get("APP_KEY")
    }
  }`

**Comment by thetutlage** [comment]
Which version of the framework?

**Reply by cassiofrancklin** [reply]
I'm using the latest versions of Adonis and Mongoose, both in their v5.    
```
    "@adonisjs/ace": "^5.0.8",
    "@adonisjs/auth": "^3.0.7",
    "@adonisjs/bodyparser": "^2.0.5",
    "@adonisjs/cors": "^1.0.7",
    "@adonisjs/fold": "^4.0.9",
    "@adonisjs/framework": "^5.0.9",
    "@adonisjs/ignitor": "^2.0.8",
    "@adonisjs/lucid": "^6.1.3",
    "mongoose": "^5.11.18"
```
